### PR TITLE
Change bool -> *bool in types.GuestCustomizationSection

### PR DESCRIPTION
**Pull request by Didainius**
In order to solve https://github.com/terraform-providers/terraform-provider-vcd/issues/303 and other problems it is better to have *bool instead of bool types (to distinguish between 0 and nil values).

This PR aims to solve just that and also deprecates some functions which are not technically correct.

Test suite passed on 9.5

### CODE REVIEW

File: govcd/vm.go [hunk 1]
```diff
@@ -459,10 +462,10 @@ func (vm *VM) Customize(computername, script string, changeSid bool) (Task, erro
 		HREF:                vm.VM.HREF,
 		Type:                types.MimeGuestCustomizationSection,
 		Info:                "Specifies Guest OS Customization Settings",
-		Enabled:             true,
+		Enabled:             takeBoolPointer(true),
```
Comment (vbauzysvmware) [inline govcd/vm.go]: Maybe it is worth it also change this harcoded value too?

File: govcd/vapp.go [hunk 2]
```diff
@@ -384,10 +387,10 @@ func (vapp *VApp) Customize(computername, script string, changeSid bool) (Task,
 		HREF:                vapp.VApp.Children.VM[0].HREF,
 		Type:                types.MimeGuestCustomizationSection,
 		Info:                "Specifies Guest OS Customization Settings",
-		Enabled:             true,
+		Enabled:             takeBoolPointer(true),
```
Comment (vbauzysvmware) [inline govcd/vapp.go]: Maybe it is worth it also change this harcoded value too?

File: govcd/vm.go (hunk 1 above)
Comment (Didainius) [reply govcd/vm.go]: This function does not have a parameter for this field, while `changeSid` had, but was missed.

File: govcd/vapp.go (hunk 2 above)
Comment (Didainius) [reply govcd/vapp.go]: This function does not have a parameter for this field, while `changeSid` had, but was missed.

File: govcd/vm.go [hunk 3]
```diff
@@ -445,6 +445,9 @@ func (vm *VM) BlockWhileGuestCustomizationStatus(unwantedStatus string, timeOutA
 	}
 }
 
+// Customize function allows to set computername, apply customization script and enable or disble the changeSid option
```
Comment (lvirbalas) [inline govcd/vm.go]: Maybe:
```suggestion
// Customize function allows to set ComputerName, apply customization script and enable or disable the changeSid option
```

Comment (Didainius) [reply govcd/vm.go]: Fixed.

**Pull request merged**